coalib/settings/ConfigurationGathering.py

```diff
@@ -386,9 +392,28 @@ def gather_configuration(acquire_settings,
                                               acquire_settings,
                                               log_printer)
     save_sections(sections)
+    add_substring_targets(targets, sections)
     warn_nonexistent_targets(targets, sections, log_printer)
 
     return (sections,
             local_bears,
             global_bears,
             targets)
+
+
+def add_substring_targets(targets, sections):
+    if not targets:
+        return
```

Comment:
Can you try the code without this? I think it will work without it as well cause `for target in targets` etc. will just iterate through an empty list.